Document: add explain analyze (#456)
risingwavelabs/risingwave-docs: sql/commands/sql-explain-analyze.mdx

---
title: "EXPLAIN ANALYZE"
description: "Use the `EXPLAIN ANALYZE` command to analyze the actual runtime performance of a streaming job."
---

> As [`EXPLAIN`](/sql/commands/sql-explain) shows the execution plan of a query or job before execution, `EXPLAIN ANALYZE` can execute the streaming job, including table, materialized view, sink, or index, and collects runtime statistics to help identify performance bottlenecks.

Reviewer: ```suggestion
As [`EXPLAIN`](/sql/commands/sql-explain) shows the execution plan of a query or job before execution, `EXPLAIN ANALYZE` analyzes a running stream job, including table, materialized view, sink, or index, and collects runtime statistics to help identify performance bottlenecks.
```

## Syntax

> To enable it, set `enable_explain_analyze_stats = true` under `[streaming.developer]` first. Set it to false will disable profiling cluster-wide.

Reviewer: It's enabled by default.

```suggestion
To disable it, set `enable_explain_analyze_stats = false` under `[streaming.developer]`.
```

```sql

> EXPLAIN ANALYZE (TABLE <name> | MATERIALIZED VIEW <name> | SINK <name> | INDEX <name> | ID <job_id>)

Reviewer: ```suggestion
EXPLAIN ANALYZE (duration_secs <secs>) [TABLE <name> | MATERIALIZED VIEW <name> | SINK <name> | INDEX <name> | ID <job_id>]
```

Reviewer: https://github.com/risingwavelabs/risingwave/pull/20966

```

> 

Reviewer: - `duration_secs`: This parameter specifies the profiling duration in seconds.

Author: Thanks and all updated

## Metrics

When you run `EXPLAIN ANALYZE`, it returns a table showing runtime statistics for each operator involved in the execution. 

| Field name                 | Description                                                                 |
|---------------------------|-----------------------------------------------------------------------------|

> | `operator_id`             | A unique identifier for each operator.                                       |

Reviewer: We removed the `operator_id` column.

| `identity`                | The name and hierarchical structure of the operator (e.g., StreamHashJoin, MergeExecutor. |
| `actor_ids`               | The IDs of concurrent actors running this operator.                         |
| `output_rps`              | The number of records output per second (records per second).               |
| `avg_output_pending_ratio`| The average ratio of pending outputs in the buffer, indicating how quickly downstream operators are consuming data. |

## Example

```sql 
CREATE TABLE t(v1 int);

CREATE MATERIALIZED VIEW m1 as 
  select * from 
    t x join t y using(v1)
      join t z using(v1) 
        join t c using(v1);

EXPLAIN ANALYZE MATERIALIZED VIEW m1;
```

Sample output:

```

>  operator_id |                    identity                    |  actor_ids  | output_rps | avg_output_pending_ratio

Reviewer: We removed the operator_id column.
